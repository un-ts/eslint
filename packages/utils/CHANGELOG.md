# eslint-plugin-utils

## 0.4.0

### Minor Changes

- [#133](https://github.com/un-ts/eslint/pull/133) [`dfabbde`](https://github.com/un-ts/eslint/commit/dfabbde34a58838d85c9b376ebd318447f6dba10) Thanks [@JounQin](https://github.com/JounQin)! - chore: bump and add flat config support

## 0.3.2

### Patch Changes

- [#83](https://github.com/un-ts/eslint/pull/83) [`6c0420c`](https://github.com/un-ts/eslint/commit/6c0420c3512a601bcccace564197ba07b8648b7b) Thanks [@JounQin](https://github.com/JounQin)! - docs: incorrect README content

## 0.3.1

### Patch Changes

- [#73](https://github.com/un-ts/eslint/pull/73) [`3516221`](https://github.com/un-ts/eslint/commit/3516221792f305a99cea28f0d2c81cce3adfba23) Thanks [@JounQin](https://github.com/JounQin)! - - chore: migrate to unts org
  - chore: add donate field, bump deps

## 0.3.0

### Minor Changes

- [`2616f9d`](https://github.com/un-ts/eslint/commit/2616f9dbf3fc81935f987907e56fa5af3f9266d3) Thanks [@JounQin](https://github.com/JounQin)! - chore(deps)!: bump all (dev)Dependencies, use exports field

## 0.2.0

### Minor Changes

- [`a2c68dd`](https://github.com/un-ts/eslint/commit/a2c68dd7ac4ff9b67b0cb1f371a008cf359dae19) Thanks [@JounQin](https://github.com/JounQin)! - feat: update entry files extensions

## 0.1.0

### Minor Changes

- [#22](https://github.com/un-ts/eslint/pull/22) [`2270c8b`](https://github.com/un-ts/eslint/commit/2270c8b0049725eb59bbc0cc45e9655b6162d735) Thanks [@JounQin](https://github.com/JounQin)! - feat: report correct ruleId from original linter
