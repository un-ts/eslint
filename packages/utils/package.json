{
  "name": "eslint-plugin-utils",
  "version": "0.4.0",
  "type": "module",
  "description": "Utilities for ESLint plugins and custom rules",
  "repository": "git+https://github.com/un-ts/eslint.git",
  "homepage": "https://github.com/un-ts/eslint/blob/main/packages/utils",
  "author": "JounQin (https://www.1stG.me) <admin@1stg.me>",
  "funding": "https://opencollective.com/unts",
  "license": "MIT",
  "engines": {
    "node": "^14.18.0 || >=16.0.0"
  },
  "main": "./lib/index.cjs",
  "module": "./lib/index.js",
  "exports": {
    "types": "./lib/index.d.ts",
    "require": "./lib/index.cjs",
    "default": "./lib/index.js"
  },
  "types": "./lib/index.d.ts",
  "files": [
    "lib"
  ],
  "keywords": [
    "eslint",
    "eslint-utils",
    "eslint-plugin-utils"
  ],
  "peerDependencies": {
    "@typescript-eslint/utils": ">=7.0.0",
    "eslint": ">=9.22.0"
  },
  "peerDependenciesMeta": {
    "@typescript-eslint/utils": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public"
  },
  "sideEffects": false
}
