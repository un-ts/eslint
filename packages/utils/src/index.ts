export * from './helpers.js'
export * from './processors.js'
export * as processors from './processors.js'
