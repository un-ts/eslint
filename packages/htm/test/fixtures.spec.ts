import path from 'node:path'

import { TSESLint } from '@typescript-eslint/utils'

import { configs } from 'eslint-plugin-htm'

const eslint = new TSESLint.ESLint({
  baseConfig: [configs.flatRecommended],
  ignore: false,
  overrideConfigFile: true,
})

test('fixtures', async () => {
  const results = await eslint.lintFiles(
    path.resolve(import.meta.dirname, 'fixtures/*.html'),
  )
  for (const { filePath, messages } of results) {
    expect(messages).toMatchSnapshot(path.basename(filePath))
  }
})
