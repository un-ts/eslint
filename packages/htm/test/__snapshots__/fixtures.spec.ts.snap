// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > basic.html 1`] = `
[
  {
    "column": 1,
    "line": 1,
    "message": "Doctype must be declared first.
evidence: <!--comment--><!DOCTYPE html>
reference: https://htmlhint.com/docs/user-guide/rules/doctype-first",
    "nodeType": null,
    "ruleId": "htm/doctype-first",
    "severity": 1,
  },
]
`;
